packages/flutter/lib/src/rendering/object.dart

```diff
@@ -548,6 +548,30 @@ class PaintingContext extends ClipContext {
     return layer;
   }
 
+  /// Blend further painting with an image filter.
+  ///
+  /// {@macro flutter.rendering.object.pushLayer.offset}
+  ///
+  /// The `imageFilter` argument is the [ImageFilter] value to use when blending
+  /// the painting done by `painter`.
+  ///
+  /// The `painter` callback will be called while the `imageFilter` is applied.
+  /// It is called synchronously during the call to [pushImageFilter].
+  ///
+  /// {@macro flutter.rendering.object.oldLayer}
+  ///
+  /// A [RenderObject] that uses this function is very likely to require its
+  /// [RenderObject.alwaysNeedsCompositing] property to return true. That informs
+  /// ancestor render objects that this render object will include a composited
+  /// layer, which, for example, causes them to use composited clips.
+  ImageFilterLayer pushImageFilter(Offset offset, ui.ImageFilter imageFilter, PaintingContextCallback painter, { ImageFilterLayer oldLayer }) {
```

Comment:
In a previous PR @Hixie asked me not to add any more methods in this class and instead have the `RenderObject` implementation instantiate the layer directly (there are several examples in `proxy_box.dart` that do that). IIRC the reasoning was that we cannot scale this class to include arbitrary custom layers that users may create. I wonder if this guideline can change now that we have extension methods that would allow users add methods to this class for their custom layers.

Reply:
Done. I see the rationale seeing as how easy it was to just use pushLayer.

Reply:
extension methods don't change this, as flar says there's really no reason to prefer doing it on the painting context.